src/components/PeopleFilters.tsx

```diff
@@ -1,21 +1,74 @@
-export const PeopleFilters = () => {
+import { useSearchParams } from 'react-router-dom';
+import React, { FC } from 'react';
+import classNames from 'classnames';
+import { SearchLink } from './SearchLink';
+
+export const PeopleFilters: FC = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  function updateSearch(params: { [key: string]: string | string[] | null }) {
+    Object.entries(params).forEach(([key, value]) => {
+      if (value === null) {
+        searchParams.delete(key);
+      } else if (Array.isArray(value)) {
+        searchParams.delete(key);
+
+        value.forEach(part => {
+          searchParams.append(key, part);
+        });
+      } else {
+        searchParams.set(key, value);
+      }
+    });
+
+    setSearchParams(searchParams);
+  }
```

Comment:
This can be split to a function helper that returns a new searchParams and then use setSearchParams to set it to the state. The component's body should stay as small and clear as possible.